### PR TITLE
Replace "master" with "coordinator" in utilities

### PR DESCRIPTION
**Pull request by jmcatamney**
This commit replaces most instances of "master" in files related to the
management utilities with "coordinator" instead, including user-facing
output, source code, test code, and documentation.  Related terms are
updated as well, such as replacing "standby master" with "standby
coordinator", "mdw" and "smdw" with "cdw" and "scdw", and so on.

Specifically, everything under gpMgmt/ is updated, as are the files in
concourse/ used by utilities tests, the Postgres utilities in src/bin/
called by management utilities, the installation and utilities guides
in gpdb-doc, and some files in src/tools/.  Backend code and anything
else unrelated to utilities remains unchanged.

Two sets of changes were *not* made in this commit.  First, to retain
backwards compatibility, flags and variables with "master" in their
names were left unchanged, such as the "--masteronly" flag in gpconfig,
the $MASTER_DATA_DIRECTORY environment variable, and "MASTER_*" entries
in gpinitsystem config files.  Second, because it would require updates
to CCP scripts, "mdw" was not changed to "cdw" in the Behave tests and 
pipeline scripts, only in unit tests and documentation.  These changes
will be addressed in later commits.

---

Notes for reviewers:
- There are a lot of changes in one commit because most of the changes were done with wide-ranging `sed` replacements and there weren't any obvious logical breaks where things could be separated out into multiple commits, except by splitting on different directories which wouldn't help with reviewing the individual files.
- I'm fairly sure I included all the changes we _do_ want to make, but catching changes we _don't_ want to make was harder.  Looking out for potential trouble spots (like e.g. a stray instance of `postmaster` becoming `postcoordinator`) would be the most helpful.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,11 +22,11 @@ no contribution is too small, we encourage all types of contributions.
 ## Overview
 
 A Greenplum cluster consists of a __master__ server, and multiple
```
Comment (kalensk) [inline README.md:24]: nit: could rename `__master__ ` to `__ coordinator__ `

Comment (jmcatamney) [reply README.md]: Good catch, fixed.

**Review by kalensk — COMMENTED** [review]
Overall looks good (from everything that I have reviewed so far...)

File: gpMgmt/bin/stream/stream.c [hunk 2]
```diff
@@ -248,7 +248,7 @@ main()
     printf(HLINE);
 #pragma omp parallel 
     {
-#pragma omp master
+#pragma omp coordinator
```
Comment (kalensk) [inline gpMgmt/bin/stream/stream.c:251]: Do we build a binary called `coordinator` now that this got renamed? Is this picking it up correctly, etc.?

**Comment by dkrieger-pvtl** [comment]
nits:
`gpMgmt/bin/gpactivatestandby`: "pid = gp.get_postmaster_pid_locally(options.coordinator_data_dir) "
                 has an extra space at the end(remove that space)
`gpMgmt/doc/*_help`: many lines you changed have a trailing " " which should be removed

**Comment by dkrieger-pvtl** [comment]
I think you want to change these two:
`gpAux/gpdemo/README`: * Master and segments TCP port:
`gpAux/gpdemo/demo_cluster.sh`: echo " This port is needed by the Master Database instance. "

**Review by dkrieger-pvtl — COMMENTED** [review]
I have given it a first pass.  Overall, it looks pretty good.  I mainly looked at negative cases as you suggested.  I did not find any flag changes for user-facing utilities.

I didn't see any figures(png) that are associated with our gpdb-doc/dita files.  But note we'll want to change them.  For instance,  `gpdb-doc/dita/graphics/standby_master.jpg` uses the master terminology as an org.  

For for user-visible input parameters, we plan to deprecate the "--master/-m"
options in favor of "--coordinator/-c" options in 7.  When this PR goes in, we'll have
a period of time with (see gpconfig_help) with things like: 
`-m | --mastervalue coordinator_value` Is this what we want?  That is, flags with `master/m` but the placeholder value with `coordinator`.

I didn't see a full pipeline, just a CM one.  It'd be good to run this on a full pipeline too as there are a lot of changes that are impossible to adequately review manually.  

For python in particular, we do not have a static checker and our code coverage isn't stellar, so I am a bit concerned there.  But since you used regex to replace, the risk of something "weird" happening is low.

**Review by kalensk — APPROVED** [review]
Looks good (assuming full pipeline goes green.)

File: gpMgmt/bin/stream/stream.c (hunk 2 above)
Comment (jmcatamney) [reply gpMgmt/bin/stream/stream.c:251]: Good catch.  That's referring to a master thread, not a GPDB master, so I've changed it back.

**Comment by jmcatamney** [comment]
@dkrieger-pvtl 
> nits:
> `gpMgmt/bin/gpactivatestandby`: "pid = gp.get_postmaster_pid_locally(options.coordinator_data_dir) "
> has an extra space at the end(remove that space)
> `gpMgmt/doc/*_help`: many lines you changed have a trailing " " which should be removed

I noticed those when doing the rename, but there are enough trailing whitespace issues in the codebase that fixing that is probably best left to another PR.

> I think you want to change these two:
> `gpAux/gpdemo/README`: * Master and segments TCP port:
> `gpAux/gpdemo/demo_cluster.sh`: echo " This port is needed by the Master Database instance. "

Fixed.

> For for user-visible input parameters, we plan to deprecate the "--master/-m"
> options in favor of "--coordinator/-c" options in 7. When this PR goes in, we'll have
> a period of time with (see gpconfig_help) with things like:
> `-m | --mastervalue coordinator_value` Is this what we want? That is, flags with master/m but the placeholder value with coordinator.

I'd say that is what we want, since the actual value is the "coordinator value" and the flag name is vestigial; I imagine future help text would look something like `-c | --coordinatorvalue <coordinator_value> OR -m | --mastervalue <coordinator_value>` to indicate that the flag is either `coordinator*` or `master*` but both point to the same thing.

> I didn't see a full pipeline, just a CM one. It'd be good to run this on a full pipeline too as there are a lot of changes that are impossible to adequately review manually.

The full pipeline is running now that the utilities-only one has gone green, and is attached to the story if anyone wants to keep an eye on it.

**Pull request merged**